server/src/main/java/org/jboss/as/server/controller/resources/ModuleInfoHandler.java

```diff
@@ -0,0 +1,209 @@
+        }
+    }
+
+
+    private ModuleLoaderMXBean getMxBean(ModuleLoader loader) throws ReflectiveOperationException {
+        Field mxBeanField = ModuleLoader.class.getDeclaredField("mxBean");
+        mxBeanField.setAccessible(true);
+        return (ModuleLoaderMXBean) mxBeanField.get(loader);
+    }
+
+    /*
+    Here is the information to expose (use ModuleLoaderMXBean as a source of information):
+
+        Loaded module names
+        For each module:
+            Module name
+            Main class name (if any)
+            Class loader name string
+            Fallback loader name string
+            Dependency information:
+                Dependency type
+                Export filter string
+                Import filter string
+                Dependency module name
+                Is-optional flag
+                Local loader type class name
+                Local loader paths
+            Resource loader information for each module:
+                Resource loader type string
+                Resource loader paths
+
+    Note that there are multiple module loaders. For our purposes there should be (at least) two categories displayed:
+
+        The static module loader
+        The deployment module loader
+
+
+     */
+    private ModelNode displayModuleInfo(ModuleInfo module) throws Exception {
+        ModelNode result = new ModelNode();
+        result.get("name").set(module.getName());
+
+        if (module.getMainClass() != null) {
+            result.get("main-class").set(module.getMainClass());
+        }
+        if (module.getFallbackLoader() != null) {
+            result.get("fallback-loader").set(module.getFallbackLoader());
+        }
+
+        ModelNode dependencies = result.get("dependencies");
+        for (DependencyInfo dependencySpec : module.getDependencies()) {
+            if (dependencySpec.getModuleName() == null) {
+                continue; //todo check why it returns empty dependancy
+            }
+            ModelNode dependency = dependencies.add();
+            dependency.get("dependency-name").set(dependencySpec.getDependencyType());
+            dependency.get("module-name").set(dependencySpec.getModuleName());
+            dependency.get("export-filter").set(dependencySpec.getExportFilter());
+            dependency.get("import-filter").set(dependencySpec.getImportFilter());
+            dependency.get("optional").set(dependencySpec.isOptional());
+            if (dependencySpec.getLocalLoader() != null) {
+                dependency.get("local-loader-class").set(dependencySpec.getLocalLoader());
+            }
+            if (dependencySpec.getLocalLoaderPaths() != null) {
+                ModelNode paths = dependency.get("local-loader-paths");
+                for (String path : dependencySpec.getLocalLoaderPaths()) {
+                    paths.add(path);
+                }
+            }
+        }
+        ModelNode resourceLoaders = result.get("resource-loaders");
```

Comment:
This should be ModelNode resourceLoaders = result.get("resource-loaders").setEmptyList(), or RESOURCE_LOADERS should be configured to allowNull.

I think the former is better.